### PR TITLE
Update all workflows to use ubuntu-22.04

### PR DESCRIPTION
**Pull request by haslinghuis**
See reminder: https://github.com/betaflight/betaflight-configurator/pull/2913

`ubuntu-22.04` seems to be finished now:

image

Need to check for Java availability: https://github.com/actions/virtual-environments/issues/5490

Has to wait until Android and Apple are happy.

### CODE REVIEW

**Comment by haslinghuis** [comment]
```
(node:4807) [DEP0128] DeprecationWarning: Invalid 'main' field in '/home/runner/work/betaflight-configurator/betaflight-configurator/dist_cordova/node_modules/objectorarray/package.json' of 'dist/index.js'. Please either fix that or report it to the module author
[20:31:11] Finished 'cordova_platforms' after 26 s
[20:31:11] Starting 'debug'...
No platform suitable for NW Build
[20:31:11] Finished 'debug' after 705 μs
[20:31:11] Starting 'cordova_debug_build'...
[20:31:11] 'cordova_debug_build' errored after 509 ms
[20:31:11] CordovaError: Failed to find 'ANDROID_HOME' environment variable. Try setting it manually.
Failed to find 'android' command in your 'PATH'. Try update your 'PATH' to include path to valid SDK directory.
    at /home/runner/work/betaflight-configurator/betaflight-configurator/dist_cordova/platforms/android/cordova/lib/check_reqs.js:305:19
    at _fulfilled (/home/runner/work/betaflight-configurator/betaflight-configurator/dist_cordova/node_modules/q/q.js:854:54)
    at /home/runner/work/betaflight-configurator/betaflight-configurator/dist_cordova/node_modules/q/q.js:883:30
    at Promise.promise.promiseDispatch (/home/runner/work/betaflight-configurator/betaflight-configurator/dist_cordova/node_modules/q/q.js:816:13)
    at /home/runner/work/betaflight-configurator/betaflight-configurator/dist_cordova/node_modules/q/q.js:877:14
    at runSingle (/home/runner/work/betaflight-configurator/betaflight-configurator/dist_cordova/node_modules/q/q.js:137:13)
    at flush (/home/runner/work/betaflight-configurator/betaflight-configurator/dist_cordova/node_modules/q/q.js:125:13)
    at processTicksAndRejections (node:internal/process/task_queues:78:11)
[20:31:11] 'debug-release' errored after 52 s
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.

```

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> FAIL
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by McGiverGim** [comment]
Yes, it seems Java is missing, and of course Android tools too (that need Java). Strange, I have Java in Ubuntu 22.04 at home without problem.
I suppose they only need some time to add it, another option is to get them via apk, but I think is better to simply wait for the moment.

**Comment by haslinghuis** [comment]
@McGiverGim did a re-run and now everything is building.

**Review by McGiverGim — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Pull request merged**
